src/Controller.ts

```diff
@@ -27,82 +29,74 @@ export default class Controller {
   }
 
   private subscribeViewEvents() {
-    addEventListener('@add', () => {
-      this.addOrder();
+    addEventListener('@add', this.addOrder.bind(this));
+    addEventListener('@submit', this.handleSubmit.bind(this));
+    addEventListener('@buttonClick', e => {
+      this.selectCoffee((e as CustomEvent).detail);
     });
-    addEventListener('@submit', () => {
-      this.handleSubmit();
+    addEventListener('@edit', e => {
+      this.editOrder((e as CustomEvent).detail);
     });
-    this.mainView.on('click', () => this.editOrder());
-    this.mainView.on('click', () => this.selectCoffee());
-    this.headerView.on('click', () => this.handleTab());
-    this.modalView.on('click', () => this.handleClose());
+  }
+
+  private bindedEvents() {
+    this.headerView.on('click', this.handleTab.bind(this));
+    this.modalView.on('click', this.handleClose.bind(this));
+  }
+
+  private selectCoffee(currentElement: HTMLButtonElement) {
+    const coffeeFilling = qs('.filling') as HTMLDivElement;
+    const coffeeName = qs('.coffee_name') as HTMLHeadingElement;
+    if (this.order.OrderItem.length === 0) {
```

Comment:
```suggestion
    if (this.order.isEmpty()) {
```
같이 객체에게 역할을 넘길수도 있을것같아요

Reply:
직관적이고 좋습니다 !